docs/contracts.rst

```diff
@@ -40,7 +40,7 @@ This means that cyclic creation dependencies are impossible.
 
 ::
 
-    pragma solidity ^0.4.16;
+    pragma solidity >0.4.21;
```

Comment:
Why `>`? `^` ensures that it will complain on 0.5.0 which may be breaking it. For 0.5.0 we'll update those pragmas at once.

Reply:
@axic: chriseth suggested using ``>`` instead - otherwise we would need to use ``pragma solidity ^0.4.21; // should actually be 0.4.22``, which would prevent those scripts from being tested by the testing scripts.

@chriseth Do you agree that we should change this back to ``^``?

Reply:
`>0.4.21` should fail the tests too as it doesn't satisfies `0.4.21`

Reply:
All tests are passing - ``>0.4.21`` does indeed fail for ``0.4.21`` (as desired, since the scripts won't run with ``0.4.21``), but it will pass for ``develop`` in between ``0.4.21`` and ``0.4.22``...

Reply:
The idea behind using `>` is that we do not have to update it until 0.5.0. I forgot about `^` also meaning `<0.6.0`. We could add that, but I'm not 100% sure that `>0.4.21 && <0.6.0` is implemented.

Reply:
Looking at https://github.com/ethereum/solidity/blob/develop/libsolidity/analysis/SemVerHandler.cpp#L178 the syntax for conjunctions would be:
``pragma solidity >0.4.21 <0.6.0;``

I just tested this locally and it seems to work - however, I'm not sure that this is the nicest solution.

To summarize, so far we have the following options:
- use ``pragma solidity ^0.4.21; // should actually be 0.4.22`` and change when version is bumped. This disables testing of the script until the comment is removed.
- use ``pragma solidity >0.4.21;`` ignoring that this does not imply ``<0.6.0``.
- use ``pragma solidity >0.4.21 <0.6.0;`` which doesn't look particularly straightforward

Another option would be to use ``pragma solidity ^0.4.22; // only works after version is bumped`` or something similar, since the scripts would also be ignored in the tests due to the coment.

I'm in favour of ``pragma solidity >0.4.21;``, since this enables testing before 0.4.22, while still not suggesting that the script works for 0.4.21, and additionally for opening an issue to change this to ``pragma solidity ^0.4.22;`` at the time of the release of 0.4.22.

Reply:
Note that `^0.4.21` does not disable testing. It is just incorrect - 0.4.21 did not have the feature we are using there and the code should better not hit the main documentation because it will be unusable (although currently it does not really hit the main documetation).

Reply:
``^0.4.21`` will not disable testing itself, but the comment ``// should actually be 0.4.22`` will (see https://github.com/ethereum/solidity/blob/develop/scripts/isolate_tests.py#L69: ``m = re.search(r'^    pragma solidity .*[0-9]+\.[0-9]+\.[0-9]+;$', l)`` expects a line end and won't match the comment).

Still can we agree that using ``>0.4.21`` plus creating an issue for changing it on the release of 0.4.22 is ok? Then I'd create the issue and this could be merged... to me so far this seems to be the best alternative.

Reply:
Oh wow, I was not aware of that. I think we should make the regex more flexible, I don't think this was the intention.

I'm fine with your suggestion.

Reply:
@axic Are you OK with the solution ``>0.4.21`` together with the issue I just created for changing it on release of ``0.4.22``?

Reply:
> All tests are passing - >0.4.21 does indeed fail for 0.4.21 (as desired, since the scripts won't run with 0.4.21), but it will pass for develop in between 0.4.21 and 0.4.22...

I forgot we are on 0.4.22.

I am confused. Why isn't `^0.4.22` working, but `>0.4.21` is?

Reply:
Because `0.4.22-develop.2018.3.13` is considered to be strictly less than `0.4.22` (it is a pre-release version).

Reply:
> Because `0.4.22-develop.2018.3.13` is considered to be strictly less than `0.4.22` (it is a pre-release version).

Ah, rather confusing to remember that :)